### PR TITLE
[SDQ-2009] Prevent disabling rubocop, except for existing exceptions in the monolith

### LINKED ISSUE #14: Rubocop is frequently being disabled in our code - start to disallow disabling Rspec
Rubocop is frequently being disabled, suggesting that our rulesets are too strict.  I suggest a series of changes to relax our rspec rules, as well as enforcing no disabling of rspec via https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/DisableCopsWithinSourceCodeDirective . 
Specific cops can be allowed with the ‘AllowedCops` configuration - I suggest starting with whitelisting all existing disables to make sure we can continue working as is, and strangling out individual disables one by one before removing from the whitelist.

### PR DESCRIPTION
**Pull request by rhunterharris**
Per https://github.com/sendoso/rubocop-sendoso/pull/17 we want to enforce disabling Lint/MissingSuper, except for certain parent classes.  As such, we've added this directive.  Following up on https://github.com/sendoso/rubocop-sendoso/issues/14. 

We have whitelisted all rubocop disable types present in the monolith here.  This conveniently provides an in-code documentation of directives we need to follow up on and disable or change the rules for.

All monolith rubocop:disable categories are allowlisted here, excluding:
Lint/MissingSuper (https://github.com/sendoso/rubocop-sendoso/pull/17/files)
Rails/SkipsModelValidations (https://github.com/sendoso/rubocop-sendoso/pull/15)

### CODE REVIEW

File: config/rubocop-style.yml [hunk 1]
```diff
@@ -20,6 +20,74 @@ Style/DefWithParentheses:
     arguments.
   Enabled: true
 
+Style/DisableCopsWithinSourceCodeDirective:
+  Description: Detects comments to enable/disable RuboCop. This is useful if want to make sure that every RuboCop error gets fixed and not quickly disabled with a comment.
+  Enabled: true
+  AllowedCops:
```
Comment (marcferna) [inline config/rubocop-style.yml:26]: so we are allowing the following cops to be disabled?

Comment (rhunterharris) [reply config/rubocop-style.yml:26]: that is correct.

Ideally we'll go through this list, relax/tighten rules as needed, then remove from this list.

Comment (gauravs) [reply config/rubocop-style.yml:26]: This looks good. Wondering if this list can be moved to https://github.com/sendoso/sendoso/blob/master/.rubocop.yml as this seems specific to that repo.

Comment (rhunterharris) [reply config/rubocop-style.yml:26]: This is located here under the assumption that other repos disable the same cops that the monolith, thus keeping this as a central source of truth for rubocop standards.

Comment (gauravs) [reply config/rubocop-style.yml:26]: Makes sense 👍🏾

**Review by gauravs — APPROVED** [review]
Looks good!

**Review by marcferna — APPROVED** [review]

**Pull request merged**